robottelo/ui/base.py

```diff
@@ -415,6 +415,14 @@ def edit_entity(self, edit_loc, edit_text_loc, entity_value, save_loc):
         self.text_field_update(edit_text_loc, entity_value)
         self.click(save_loc)
 
+    def set_limit(self, limit):
+        """Specify content host limit value for host collection or activation
+        key entities.
+        """
+        self.click(locators['hostcollection.usage_limit_checkbox'])
+        if limit != 'Unlimited':
+            self.field_update('hostcollection.usage_limit', limit)
+
```

Comment:
your argument is partially valid.  Did you also see that all those functions you mentioned use `common_locators` and not entity specific locators?  This explains that these functions are written in a generic way to suit any entities.  But your case is not. If you want to make it generic - then :+1: